gandalf.js

```diff
@@ -1,20 +1,18 @@
 /**
```

Comment:
SCRIPT-I'm impressed with your use of design patterns here, it really enhances the code organization.